File: PracticaExamen/ObjetosTransferencia/TOFactura.vb
Public Class TOFactura
    Property ID_Factura As Integer
    Property Fecha As Date
    Property Cliente As TOCliente
    Property Detalle As TOListaProductos
    Property Total As Integer

    Sub New()

    End Sub

    Sub New(id As Integer, f As Date, c As TOCliente, d As TOListaProductos, t As Integer)
        Me.ID_Factura = id
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = d
        Me.Total = t
    End Sub

    Sub New(id As Integer, f As Date, c As TOCliente, t As Integer)
        Me.ID_Factura = id
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = New TOListaProductos()
        Me.Total = t
    End Sub

    Sub New(f As Date, c As TOCliente, d As TOListaProductos, t As Integer)
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = d
        Me.Total = t
    End Sub

    Sub New(f As Date, c As TOCliente, t As Integer)
        Me.Fecha = f
        Me.Cliente = c
        Me.Detalle = New TOListaProductos()
        Me.Total = t
    End Sub
End Class

File: Practicav2/ObjetosTransferencia/TOFactura.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ObjetosTransferencia
{
    public class TOFactura
    {
        public int Numero { get; set; }
        public DateTime Fecha { get; set; }
        public TOCliente ClienteF { get; set; }
        public TODetalleFactura detalle { get; set; }
        public int Total { get; set; }

        public TOFactura() { }

        public TOFactura(int num, DateTime fecha, TOCliente c, TODetalleFactura d, int t)
        {
            this.Numero = num;
            this.Fecha = fecha;
            this.ClienteF = c;
            this.detalle = d;
            this.Total = t;
        }
    }
}
